src/databricks/sql/client.py

```diff
@@ -225,6 +228,26 @@ def __del__(self):
     def get_session_id(self):
         return self.thrift_backend.handle_to_id(self._session_handle)
 
+    @staticmethod
+    def get_protocol_version(openSessionResp):
+        if (
```

Comment:
Since this is a new behaviour to fetch first from the `sessionHandle` versus falling back to the `openSessionResp`, please add a docstring or comment explaining why we do this.